test/fixtures/azure-fixture/src/main/java/fixture/azure/MockAzureBlobStore.java

```diff
@@ -0,0 +1,483 @@
+     */
+    public static class Lease {
+
+        /**
+         * Minimal set of states, we don't support breaking/broken
+         */
+        enum State {
+            Available,
+            Leased,
+            Expired
+        }
+
+        private String leaseId;
+        private long expireTimeMillisSinceEpoch = Long.MAX_VALUE;
+        private State state = State.Available;
+
+        public synchronized String acquire(@Nullable String proposedLeaseId, int leaseTimeSeconds) {
+            expireIfDue();
+            switch (state) {
+                case Available, Expired -> {
+                    final State prevState = state;
+                    state = State.Leased;
+                    leaseId = proposedLeaseId != null ? proposedLeaseId : UUID.randomUUID().toString();
+                    setExpireTimeMillisSinceEpoch(leaseTimeSeconds);
+                    logger.debug(
+                        "Granting lease, prior state={}, leaseId={}, expires={}",
+                        prevState,
+                        leaseId,
+                        Instant.ofEpochMilli(expireTimeMillisSinceEpoch)
+                    );
+                }
+                case Leased -> {
+                    if (leaseId.equals(proposedLeaseId) == false) {
+                        logger.debug("Mismatch on acquire - proposed leaseId: {}, active leaseId: {}", proposedLeaseId, leaseId);
+                        throw new ConflictException(
+                            "LeaseIdMismatchWithLeaseOperation",
+                            "The lease ID specified did not match the lease ID for the blob/container."
+                        );
+                    }
+                    setExpireTimeMillisSinceEpoch(leaseTimeSeconds);
+                }
+            }
+            return leaseId;
+        }
+
+        public synchronized void release(String requestLeaseId) {
+            expireIfDue();
+            switch (state) {
+                case Available -> throw new ConflictException(
+                    "LeaseNotPresentWithLeaseOperation",
+                    "There is currently no lease on the blob/container."
+                );
+                case Leased, Expired -> {
+                    if (leaseId.equals(requestLeaseId) == false) {
+                        logger.debug("Mismatch on release - submitted leaseId: {}, active leaseId: {}", requestLeaseId, this.leaseId);
+                        throw new ConflictException(
+                            "LeaseIdMismatchWithLeaseOperation",
+                            "The lease ID specified did not match the lease ID for the blob/container."
+                        );
+                    }
+                    state = State.Available;
+                    this.leaseId = null;
+                    this.expireTimeMillisSinceEpoch = Long.MAX_VALUE;
+                }
+            }
+        }
+
+        public synchronized void assertCanWrite(@Nullable String requestLeaseId) {
```

Comment:
Could we keep the term `assert` to mean "check for logic errors and fail the test" rather than just checking a precondition?

Reply:
Addressed in 5a47eeda9de9d366453d372d7f2cd1f59c1e9bb5

Reply:
Doesn't seem to have fixed it on this branch?

Reply:
Oops I missed some. Remaining asserts gone in 0a72bc94a2242d1281afa1bcceb6d2a8cd6556df